includes/admin/class-wc-rest-payments-customer-controller.php

```diff
@@ -0,0 +1,347 @@
+<?php
+/**
+ * Class WC_REST_Payments_Customer_Controller
+ *
+ * @package WooCommerce\Payments\Admin
+ */
+
+use WCPay\Core\Server\Request;
+use WCPay\Exceptions\API_Exception;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * REST controller for customers.
+ */
+class WC_REST_Payments_Customer_Controller extends WC_Payments_REST_Controller {
+
+	/**
+	 * Onboarding Service.
+	 *
+	 * @var WC_Payments_Customer_Service
+	 */
+	protected $customer_service;
+
+	/**
+	 * Endpoint path.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'payments/customer';
+
+	/**
+	 * Constructor.
+	 *
+	 * @param WC_Payments_API_Client       $api_client    WooCommerce Payments API client.
+	 * @param WC_Payments_Customer_Service $customer_service Token service.
+	 */
+	public function __construct(
+		WC_Payments_API_Client $api_client,
+		WC_Payments_Customer_Service $customer_service
+	) {
+		parent::__construct( $api_client );
+		$this->customer_service = $customer_service;
+	}
+
+	/**
+	 * Configure REST API routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<customer_id>\d+)/payment_methods',
+			[
+				[
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => [ $this, 'get_customer_payment_methods' ],
+					'permission_callback' => [ $this, 'check_permission' ],
+					'args'                => $this->get_collection_params(),
+				],
+				'schema' => [ $this, 'get_item_schema' ],
+			]
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<customer_id>\d+)/payment_methods/(?P<payment_method_id>\w+)',
+			[
+				[
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => [ $this, 'get_single_customer_payment_methods' ],
+					'permission_callback' => [ $this, 'check_permission' ],
+				],
+				'schema' => [ $this, 'get_item_schema' ],
+			]
+		);
+	}
+
+	/**
+	 * Retrieve transaction to respond with via API.
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 */
+	public function get_customer_payment_methods( $request ) {
+		$payment_customer_id   = $this->customer_service->get_customer_id_by_user_id( $request->get_param( 'customer_id' ) );
```

Comment:
This line is not working, we are considering `customer_id` as `user_id`

Reply:
I used WooCommerce user it and not stripe customer id to keep it close to the WooCommerce REST API. Do you think that we need to change that and introduce new customer into the play?

Reply:
Fixed by switching to customer id, since user id not available on metadata.